ark/mibi/tiling_utils.py

```diff
@@ -82,69 +93,195 @@ def generate_region_info(region_start_x, region_start_y, fov_num_x, fov_num_y,
+        region_rand (list):
+            Whether to set randomization or not sorted by run
     """
 
-    # file path validation
-    if not os.path.exists(fov_list_path):
-        raise FileNotFoundError("FOV region file %s does not exist" % fov_list_path)
+    # there has to be a starting and ending fov for each region
+    if len(fov_tile_info['fovs']) % 2 != 0:
+        raise ValueError(
+            "Data in fov_list_path needs to contain a start and end fov for each region"
+        )
 
-    if not os.path.exists(moly_path):
-        raise FileNotFoundError("Moly point file %s does not exist" % moly_path)
+    # every two fovs should define the start and end of the fov
+    for i in range(0, len(fov_tile_info['fovs']), 2):
+        # define the current start and end fov
+        fov_batches = fov_tile_info['fovs'][i:i + 2]
+        start_fov = fov_batches[0]
+        end_fov = fov_batches[1]
+
+        # the names of the fovs need to match
+        if start_fov['name'] != end_fov['name']:
+            err_msg = "FOV mismatch at indices %d, %d: names %s, %s are not equal"
+            raise ValueError(
+                err_msg % (i, i + 1, start_fov['name'], end_fov['name'])
+            )
+
+        # define the start and end coordinates
+        start_fov_x = start_fov['centerPointMicrons']['x']
+        end_fov_x = end_fov['centerPointMicrons']['x']
+        start_fov_y = start_fov['centerPointMicrons']['y']
+        end_fov_y = end_fov['centerPointMicrons']['y']
+
+        # the coordinates have to be valid
+        if start_fov_x > end_fov_x or start_fov_y > end_fov_y:
+            err_msg = ("Coordinate error at indices %d, %d: start coordinates cannot be",
+                       " greater than end coordinates")
+            raise ValueError(err_msg % (i, i + 1))
+
+        region_start_x.append(start_fov_x)
+        region_start_y.append(start_fov_y)
+        region_end_x.append(end_fov_x)
+        region_end_y.append(end_fov_y)
+
+        # the num_x, num_y, size_x, and size_y need additional validation
+        # since they may not be compatible with the starting and ending coordinates
+        while True:
+            # allow the user to specify the number of fovs along each dimension
+            num_x = read_tiling_param(
+                "Enter number of x fovs for region %s: " % start_fov['name'],
+                "Error: number of x fovs must be positive",
+                lambda nx: nx >= 1,
+                dtype=int
+            )
+
+            num_y = read_tiling_param(
+                "Enter number of y fovs for region %s: " % start_fov['name'],
+                "Error: number of y fovs must be positive",
+                lambda ny: ny >= 1,
+                dtype=int
+            )
+
+            # allow the user to specify the step size along each dimension
+            size_x = read_tiling_param(
+                "Enter the x step size for region %s: " % start_fov['name'],
```

Comment:
This text needs to be updated for the TMA script. It should say image size, not step size